### PR TITLE
Updated flash emulation...

### PR DESCRIPTION
**Pull request by mikestaib**
TL;DR - This code should provide a more-or-less accurate emulation of the mode 7 data using mode 8 (as well as the f125 long and short modes). 

Updated the f125 and f250 firmware emulation scheme. This restructuring removes the bulk of the code from JEventSource_EVIO that was doing the emulation and moves fa125_algos to a new class. The new code uses tagged classes of the type Df125EmulatorAlgorithm and Df250EmulatorAlgorithm to emulate the firmware when raw data is available. In the future, modeling a new algorithm can be achieved by making a new tagged EmulatorAlgorithm factory and using the DEFTAG feature of JANA.
- NEW: The parameters used for the emulation are now retrieved from the BOR records for each of the readout channels when these records are available. In this way, the emulation of the firmware, including the number of samples in the integral are correct for each of the different subsystems according to the registers written into the flash at the beginning of the run. 
- NEW: The emulated values are saved into the raw level objects. For example, Df250PulseTime now has two members, 'time' and 'time_emulated'. If the 'emulated' flag in the raw data word is set to 1,  'time' is just a copied value of 'time_emulated'. However if the 'emulated' flag is set to 0 (meaning that the word came directly from the flash), 'time' contains the value returned by the flash, while 'time_emulated' contains the value returned from the emulation. This can be very handy for debugging new versions of the firmware.

-Note - I have not tested this code extensively with data files that predate BOR records being available in the data stream, so if you are emulating with old data files YMMV. The code should revert to the default values if the BORConfig objects are not found.
- There are still many features that can be added. If you have something in mind or run into problems, please let me know.

### CODE REVIEW

**Comment by gluex** [comment]
Test status for this pull request: SUCCESS

  Summary: [/work/halld/pull_request_test/sim-recon^FlashEmulation/tests/summary.txt](https://halldweb.jlab.org/pull_request_test/sim-recon^FlashEmulation/tests/summary.txt)
 Logs: [/work/halld/pull_request_test/sim-recon^FlashEmulation/tests/log](https://halldweb.jlab.org/pull_request_test/sim-recon^FlashEmulation/tests/log)

 Build log: [/work/halld/pull_request_test/sim-recon^FlashEmulation/make_FlashEmulation.log](https://halldweb.jlab.org/pull_request_test/sim-recon^FlashEmulation/make_FlashEmulation.log)
 Build report: [/work/halld/pull_request_test/sim-recon^FlashEmulation/report_FlashEmulation.txt](https://halldweb.jlab.org/pull_request_test/sim-recon^FlashEmulation/report_FlashEmulation.txt)
 Location of build: /work/halld/pull_request_test/sim-recon^FlashEmulation
